pkg/kudoctl/util/repo/repo.go

```diff
@@ -0,0 +1,52 @@
+package repo
+
+import (
+	"github.com/kudobuilder/kudo/pkg/apis/kudo/v1alpha1"
+)
+
+// Entry represents a collection of parameters for framework repository
+type Entry struct {
+	Name      string `json:"name"`
+	LocalPath string `json:"localPath"`
+	URL       string `json:"url"`
+}
+
+// Metadata for a Framework. This models the structure of a bundle.yaml file.
+type Metadata struct {
+	// The name of the framework
+	Name string `json:"name,omitempty"`
+	// A SemVer 2 conformant version string of the framework
+	Version string `protobuf:"bytes,4,opt,name=version" json:"version,omitempty"`
+	// The base version of the application enclosed inside of this framework.
+	BaseVersion string `json:"baseVersion,omitempty"`
+	// The KUDO API Version of this framework.
+	APIVersion string `json:"apiVersion,omitempty"`
```

Comment:
what is the api version here? It's a version of the metadata file? should we call it MetadataFileVersion then? KUDO API just sounds too generic to me

Reply:
I'd call this KudoVersion. This and KubernetesVersion should be semver ranges, so a KudoVersion can always encapsulate different file format versions by setting the right semver.

Reply:
Exactly, it should be SemVer Kudoversion under which this particular Framework will function.